src/main/java/edu/utdallas/davisbase/host/Host.java

```diff
@@ -82,48 +144,118 @@ else if (result instanceof UpdateResult) {
   }
 
   public void write(CreateIndexResult result) throws IOException {
-    // TODO Implement Host.write(CreateIndexResult)
-    throw new NotImplementedException();
+
+    try {
+      System.out.println("Index was successfully created on column '"+result.getColumnName()
+      +"' in table '"+result.getTableName()+ "'.");
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
   }
 
   public void write(CreateTableResult result) throws IOException {
-    // TODO Implement Host.write(CreateTableResult)
-    throw new NotImplementedException();
+
+    try {
+      System.out.println("'"+result.getTableName()+"' table was successfully created.");
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
   }
 
   public void write(DeleteResult result) throws IOException {
-    // TODO Implement Host.write(DeleteResult)
-    throw new NotImplementedException();
+
+    try {
+      System.out.println(result.getRowsDeleted() +" rows were deleted in the table '"+result.getTableName()+"'.");
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
   }
 
   public void write(DropTableResult result) throws IOException {
-    // TODO Implement Host.write(DropTableResult)
-    throw new NotImplementedException();
+
+    try {
+      System.out.println("'"+result.getTableName() +"' table is deleted from database.");
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
   }
 
   public void write(ExitResult result) throws IOException {
-    // TODO Implement Host.write(ExitResult)
-    throw new NotImplementedException();
+
+    try {
+      System.out.println("Exiting database....");
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
   }
 
   public void write(InsertResult result) throws IOException {
-    // TODO Implement Host.write(InsertResult)
-    throw new NotImplementedException();
+
+    try {
+      System.out.println(result.getRowsInserted() +" rows were inserted in the table '"+result.getTableName()+"'.");
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
   }
 
   public void write(SelectResult result) throws IOException {
-    // TODO Implement Host.write(SelectResult)
-    throw new NotImplementedException();
+
+    if(result.getData().size() == 0)
+      System.out.println("Empty result set.");
+
+    else {
+
+      System.out.print(line_chars("-", ((result.getSchema().size())*8)+ 3));
+      System.out.println();
+
+      for(int i = 0; i < result.getSchema().size(); i++){
+
+        System.out.print(display_fix(result.getSchema().getColumnName(i).length(), result.getSchema().getColumnName(i))+"|");
+      }
+      System.out.println();
+
+      System.out.print(line_chars("-", ((result.getSchema().size())*8)+ 3));
+      System.out.println();
+
+      for(SelectResultDataRow row : result.getData()){
+
+        while(row.iterator().hasNext()){
+
+          System.out.print(display_fix(10,row.iterator().next().toString())+"|");
```

Comment:
This isn't how iterators work. You should be using a foreach loop, and accessing the loop variable here.